Use async inventory for graph refresh

borod108 opened the pull request:
When doing full graph refresh use the aync batching to get the inventory from the provider.
This addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1507773

borod108 commented:
@pkliczewski @masayag @jhernand can you please review?
Please note I might remove some duplicated code in app/models/manageiq/providers/redhat/inventory/parser/infra_manager/full.rb (this is the wip part)

pkliczewski commented on app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb:
```diff
@@ -13,6 +19,10 @@ def parse
   end
 
+  def targeted_collection?
+    collector.class.name.include?("TargetCollection")
```
Is there any other way to distinguish the collectors?

borod108 replied:
well I could do collector.class == ...::TargetCollection, would that be better?

pkliczewski replied:
Maybe we could add a method to both

pkliczewski commented on app/models/manageiq/providers/redhat/inventory/parser/infra_manager/full.rb:
```diff
@@ -0,0 +1,587 @@
+module ManageIQ::Providers::Redhat::Inventory::Parser::InfraManager::Full
```
Can we share the code? Code climate complains about having the same code.

pkliczewski replied:
I may become our main refresh so maybe it would be good to make it a bit cleaner

pkliczewski commented on spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_4_1_spec.rb:
```diff
@@ -1,787 +0,0 @@
-describe ManageIQ::Providers::Redhat::InfraManager::Refresh::Refresher do
```
Why do we want to rename this file? I am not able to say what are the differences.

borod108 replied:
I am not renaming it, I am deleting it. We now have a new test file that does not have all those allow_to_receive_and_return things, instead it uses the OvirtSDK::VCR

pkliczewski replied:
OK

masayag commented on app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb:
```diff
@@ -163,6 +173,11 @@ def hosts
     end
   end
 
+
```
why so many new lines ?

borod108 commented:
@pkliczewski @masayag @jhernand I have refactored this PR, I moved all the difference in collecting into the collector which makes much more sense.
Can you please review?

borod108 commented:
@miq-bot remove-label wip

pkliczewski commented:
@masayag @oourfali merge?

borod108 commented:
@pkliczewski hi Moti is still reviewing :)

oourfali commented:
@miq-bot add-label gaprindashvili/yes

simaishi commented:
Gaprindashvili backport details:

```
$ git log -1
commit 86ba3e1b29aefd60a85738e2b799945d4e6d2191
Author: Moti Asayag <masayag@redhat.com>
Date:   Thu Nov 2 13:57:33 2017 +0200

    Merge pull request #121 from borod108/async_graph
    
    Use async inventory for graph refresh
    (cherry picked from commit 35522f6b2f7cde164dfae9a87834916bafd1ab29)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1510205
```
